proto/sync/sync.proto

```diff
@@ -4,26 +4,90 @@ package sync;
 
 option go_package = "github.com/ava-labs/avalanchego/proto/pb/sync";
 
+// Request represents a request for information during syncing.
 message Request {
   oneof message {
     RangeProofRequest range_proof_request = 1;
     ChangeProofRequest change_proof_request = 2;
   }
 }
 
+// A RangeProofRequest requests the key-value pairs in a given key range
+// at a specific revision.
 message RangeProofRequest {
-  bytes root = 1;
-  bytes start = 2;
-  bytes end = 3;
+  bytes root_hash = 1;
+  bytes start_key = 2;
+  bytes end_key = 3;
   uint32 key_limit = 4;
   uint32 bytes_limit = 5;
 }
 
+// A ChangeProofRequest requests the changes between two revisions.
 message ChangeProofRequest {
-  bytes start_root = 1;
-  bytes end_root = 2;
-  bytes start = 3;
-  bytes end = 4;
+  bytes start_root_hash = 1;
+  bytes end_root_hash = 2;
+  bytes start_key = 3;
+  bytes end_key = 4;
   uint32 key_limit = 5;
   uint32 bytes_limit = 6;
 }
+
+// KeyValue represents a single key and its value.
+message KeyValue {
+  bytes key = 1;
+  bytes value = 2;
+}
+
+// KeyChange is a change for a key from one revision to another.
+// If the value is None, the key was deleted.
+message KeyChange {
+  bytes key = 1;
+  MaybeBytes value = 2;
```

Comment:
This should use the `optional` keyword.

Reply:
Discussed offline. `optional` means the `[]byte` will be `nil` if not provided, which is annoying because we want to distinguish `None` from `[]byte{}`/`nil`, and we treat the two as being equivalent in other places. Using `optional` would mean that we need to explicitly use `nil`, not `[]byte{}`, when encoding.